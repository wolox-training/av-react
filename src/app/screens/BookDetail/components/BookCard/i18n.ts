import i18next from 'i18next';

i18next.addResources('es', 'BookCard', {
  back: 'Atrás',
  backArrowAlt: 'backArrowAlt',
  bookAlt: 'bookImg',
  badgeAlt: 'badgeAlt',
  author: 'Autor del libro:',
  editor: 'Editorial:',
  publishYearh: 'Año de publicación:'
});
