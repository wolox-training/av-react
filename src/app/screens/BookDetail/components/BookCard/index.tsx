import React from 'react';
import i18next from 'i18next';
import { useHistory } from 'react-router';

import { BookSuccess } from '~utils/types';

import BackArrow from '../../Assets/back-arrow.png';
import Badge from '../../Assets/badge.png';

import styles from './styles.module.scss';

interface BookProps {
  data?: BookSuccess;
}

export default function BookData({ data }: BookProps) {
  const history = useHistory();

  return (
    <div>
      <div className={`${styles.backWrapper} row middle`} onClick={() => history.goBack()}>
        <img
          src={BackArrow}
          alt={i18next.t('BookCard:backArrowAlt') as string}
          className={styles.backArrow}
        />
        <span className={styles.backText}>{i18next.t('BookCard:back')}</span>
      </div>
      <div className={`${styles.bookCard} row`}>
        <img src={data?.imageUrl} alt={i18next.t('BookCard:bookAlt') as string} className={styles.bookImg} />
        <img src={Badge} alt={i18next.t('BookCard:badgeAlt') as string} className={styles.badge} />
        <div className={`${styles.bookInfo} column`}>
          <div className={styles.titleWrapper}>
            <span className={styles.bookTitle}>{data?.title}</span>
            <span className={styles.bookGenre}>{data?.genre}</span>
          </div>
          <div className={styles.bookDetails}>
            <span className={styles.bookDetailsIndex}>{i18next.t('BookCard:author')}</span>
            <span className={styles.bookDetailsItem}>{data?.author}</span>
          </div>
          <div className={styles.bookDetails}>
            <span className={styles.bookDetailsIndex}>{i18next.t('BookCard:editor')}</span>
            <span className={styles.bookDetailsItem}>{data?.editor}</span>
          </div>
          <div className={styles.bookDetails}>
            <span className={styles.bookDetailsIndex}>{i18next.t('BookCard:publishDate')}</span>
            <span className={styles.bookDetailsItem}>{data?.year}</span>
          </div>
        </div>
      </div>
    </div>
  );
}
