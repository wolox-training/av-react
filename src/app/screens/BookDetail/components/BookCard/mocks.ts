export const OK_BOOK_RESPONSE_API = {
  id: 1,
  author: 'author',
  title: 'title',
  imageUrl: 'imageurl',
  editor: 'editor',
  year: 1900,
  genre: 'genre',
  currentRent: false
};
