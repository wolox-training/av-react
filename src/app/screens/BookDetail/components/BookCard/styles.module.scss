@import 'variables/colors';

.back-wrapper {
  cursor: pointer;

  .back-arrow {
    height: 20px;
    width: 20px;
  }

  .back-text {
    color: $black;
    font-size: 20px;
    height: 20px;
    line-height: 24px;
    margin-left: 2px;
    width: 59px;
  }
}

.book-card {
  box-shadow: 5px 5px 10px 6px $silver;
  height: 416px;
  margin-top: 108px;
  width: 900px;

  .book-img {
    flex-shrink: 0;
    height: 368px;
    margin: 24px 0 24px 24px;
    width: 261px;
  }

  .badge {
    height: 118.57px;
    margin: 1.73px 0 0 223.73px;
    position: absolute;
    width: 94.53px;
  }

  .book-info {
    margin: 24px 0 0 52px;
    width: 100%;
  }

  .title-wrapper {
    &::after {
      content: '';
      background-color: $earls-green;
      display: block;
      margin: 15px 24px 0 0;
      height: 4px;
    }
  }

  .book-title {
    color: $black;
    font-size: 32px;
    font-weight: 700;
    line-height: 38px;
  }

  .book-genre {
    color: $gray;
    font-size: 24px;
    font-weight: 700;
    line-height: 28px;
    margin-left: 8px;
  }

  .book-details {
    margin-top: 36px;
  }

  .book-details-index {
    font-size: 20px;
    font-weight: 700;
    line-height: 24px;
  }

  .book-details-item {
    color: $gray;
    font-size: 20px;
    line-height: 24px;
    margin-left: 8px;
  }
}
