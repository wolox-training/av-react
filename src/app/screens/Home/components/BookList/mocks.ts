export const OK_BOOKS_API_RESPONSE = {
  count: 1,
  currentPage: 1,
  nextPage: 1,
  page: [
    {
      id: 1,
      author: 'test author',
      title: 'test title',
      imageUrl: 'test url',
      publisher: 'test publisher',
      year: 1960,
      genre: 'test genre',
      createdAt: '2020-10-10',
      updatedAt: '2020-10-10'
    },
    {
      id: 2,
      author: 'test author 2',
      title: 'test title 2',
      imageUrl: 'test url 2',
      publisher: 'test publisher 2',
      year: 1960,
      genre: 'test genre 2',
      createdAt: '2020-10-10',
      updatedAt: '2020-10-10'
    }
  ],
  totalCount: 1,
  totalPages: 1
};
