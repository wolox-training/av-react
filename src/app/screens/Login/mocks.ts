export const OK_LOGIN_API_RESPONSE = {
  data: {
    'access-token': 'test',
    data: {
      email: 'test@test.com',
      firstName: 'test',
      id: 1,
      lastName: 'test',
      locale: 'es'
    }
  }
};
