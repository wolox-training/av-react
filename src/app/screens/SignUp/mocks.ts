export const OK_SIGNUP_API_RESPONSE = {
  headers: {
    'access-token': 'test'
  },
  data: {
    email: 'test@test.com',
    firstName: 'test',
    id: 1,
    lastName: 'test',
    locale: 'es'
  },
  ok: true,
  originalError: null,
  problem: null,
  status: 201
};
