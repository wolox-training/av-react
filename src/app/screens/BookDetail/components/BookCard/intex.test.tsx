import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';

import { OK_BOOK_RESPONSE_API } from './mocks';

import BookCard from './';

describe('BookDetail test', () => {
  test('Should match snapshot', () => {
    const { container } = render(
      <Router>
        <BookCard data={OK_BOOK_RESPONSE_API} />
      </Router>
    );
    expect(container).toMatchSnapshot();
  });

  test('Should render alt texts', () => {
    render(
      <Router>
        <BookCard data={OK_BOOK_RESPONSE_API} />
      </Router>
    );
    expect(screen.getByAltText('BookCard:backArrowAlt')).toBeInTheDocument();
    expect(screen.getByAltText('BookCard:bookAlt')).toBeInTheDocument();
    expect(screen.getByAltText('BookCard:badgeAlt')).toBeInTheDocument();
  });

  test('Should render book data', () => {
    render(
      <Router>
        <BookCard data={OK_BOOK_RESPONSE_API} />
      </Router>
    );
    expect(screen.getByText(OK_BOOK_RESPONSE_API.title)).toBeInTheDocument();
    expect(screen.getByText(OK_BOOK_RESPONSE_API.genre)).toBeInTheDocument();
    expect(screen.getByText(OK_BOOK_RESPONSE_API.author)).toBeInTheDocument();
    expect(screen.getByText(OK_BOOK_RESPONSE_API.editor)).toBeInTheDocument();
    expect(screen.getByText(OK_BOOK_RESPONSE_API.year)).toBeInTheDocument();
  });
});
