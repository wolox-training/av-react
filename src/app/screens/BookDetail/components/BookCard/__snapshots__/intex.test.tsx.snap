// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetail test Should match snapshot 1`] = `
<div>
  <div>
    <div
      class="backWrapper row middle"
    >
      <img
        alt="BookCard:backArrowAlt"
        class="backArrow"
        src="back-arrow.png"
      />
      <span
        class="backText"
      >
        BookCard:back
      </span>
    </div>
    <div
      class="bookCard row"
    >
      <img
        alt="BookCard:bookAlt"
        class="bookImg"
        src="imageurl"
      />
      <img
        alt="BookCard:badgeAlt"
        class="badge"
        src="badge.png"
      />
      <div
        class="bookInfo column"
      >
        <div
          class="titleWrapper"
        >
          <span
            class="bookTitle"
          >
            title
          </span>
          <span
            class="bookGenre"
          >
            genre
          </span>
        </div>
        <div
          class="bookDetails"
        >
          <span
            class="bookDetailsIndex"
          >
            BookCard:author
          </span>
          <span
            class="bookDetailsItem"
          >
            author
          </span>
        </div>
        <div
          class="bookDetails"
        >
          <span
            class="bookDetailsIndex"
          >
            BookCard:editor
          </span>
          <span
            class="bookDetailsItem"
          >
            editor
          </span>
        </div>
        <div
          class="bookDetails"
        >
          <span
            class="bookDetailsIndex"
          >
            BookCard:publishYearh
          </span>
          <span
            class="bookDetailsItem"
          >
            1900
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
